src/appleseed.shaders/src/maya/as_maya_granite.osl

```diff
@@ -0,0 +1,322 @@
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+//
+
+// Ref: $MAYA_LOCATION/docs/Nodes/granite.html
+
+#include "appleseed/fractal/as_fractal_helpers.h"
+#include "appleseed/math/as_math_helpers.h"
+#include "appleseed/maya/as_maya_fractal_helpers.h"
+#include "appleseed/maya/as_maya_helpers.h"
+#include "appleseed/maya/as_maya_transform_helpers.h"
+
+shader as_maya_granite
+[[
+    string as_maya_node_name = "granite",
+]]
+(
+    color in_color1 = color(0)
+    [[
+        string as_maya_attribute_name = "color1",
+        string as_maya_attribute_name_short = "c1",
+        string label = "Color 1",
+        string page = "Granite Attributes"
+    ]],
+    color in_color2 = color(0.549, 0.784, 0.392) 
+    [[
+        string as_maya_attribute_name = "color2",
+        string as_maya_attribute_name_short = "c2",
+        string label = "Color 2",
+        string page = "Granite Attributes"
+    ]],
+    color in_color3 = color(0.627, 0.824, 0.823) 
+    [[
+        string as_maya_attribute_name = "color3",
+        string as_maya_attribute_name_short = "c3",
+        string label = "Color 3",
+        string page = "Granite Attributes"
+    ]],
+    color in_fillerColor = color(0.588, 0.294, 0.196)
+    [[
+        string as_maya_attribute_name = "fillerColor",
+        string as_maya_attribute_name_short = "fc",
+        string label = "Filler Color",
+        string page = "Granite Attributes"
+    ]],
+    float in_cellSize = 0.15
+    [[
+        string as_maya_attribute_name = "cellSize",
+        string as_maya_attribute_name_short = "cs",
+        float min = 0.0,
+        float softmax = 1.0,
+        string label = "Cell Size",
+        string page = "Granite Attributes"
+    ]],
+    float in_density = 1.0
+    [[
+        string as_maya_attribute_name = "density",
+        string as_maya_attribute_name_short = "dy",
+        float min = 0.0,
+        float softmax = 1.0,
+        string label = "Density",
+        string page = "Granite Attributes"
+    ]],
+    float in_mixRatio = 0.5
+    [[
+        string as_maya_attribute_name = "mixRatio",
+        string as_maya_attribute_name_short = "mr",
+        float min = 0.0,
+        float max = 1.0,
+        string label = "Mixing Ratio",
+        string page = "Granite Attributes"
+    ]],
+    float in_randomness = 1.0
+    [[
+        string as_maya_attribute_name = "randomness",
+        string as_maya_attribute_name_short = "ra",
+        float min = 0.0,
+        float softmax = 1.0,
+        string label = "Randomness",
+        string page = "Granite Attributes"
+    ]],
+    float in_spottyness = 0.3
+    [[
+        string as_maya_attribute_name = "spottyness",
+        string as_maya_attribute_name_short = "s",
+        float min = 0.0,
+        float softmax = 1.0,
+        string label = "Spottyness",
```

Comment:
I think so, that the right term is *spottiness*, but i was following Maya's nomenclature, bugs quirks and all :)